util/fipstools/acvp/acvptool/subprocess/ml_dsa.go

```diff
@@ -0,0 +1,239 @@
+				PK: pk,
+				SK: sk,
+			})
+		}
+
+		responses = append(responses, response)
+	}
+
+	return responses, nil
+}
+
+type mlDsaSigGenTestGroup struct {
+	ID                 uint64 `json:"tgId"`
+	Type               string `json:"testType"`
+	ParameterSet       string `json:"parameterSet"`
+	Deterministic      bool   `json:"deterministic"`
+	SignatureInterface string `json:"signatureInterface"`
+	ExternalMu         bool   `json:"externalMu`
+	Tests         []struct {
+		ID      uint64               `json:"tcId"`
+		Message hexEncodedByteString `json:"message"`
+		MU      hexEncodedByteString `json:"mu"`
+		SK      hexEncodedByteString `json:"sk"`
+		RND     hexEncodedByteString `json:"rnd"`
+		Context hexEncodedByteString `json:"context"`
+		HashAlg string               `json:"hashAlg"`
+	}
+}
+
+type mlDsaSigGenTestGroupResponse struct {
+	ID    uint64                        `json:"tgId"`
+	Tests []mlDsaSigGenTestCaseResponse `json:"tests"`
+}
+
+type mlDsaSigGenTestCaseResponse struct {
+	ID        uint64               `json:"tcId"`
+	Signature hexEncodedByteString `json:"signature"`
+}
+
+// Convert boolean to byte slice (using 1 for true, 0 for false)
+func boolToBytes(b bool) []byte {
+    if b {
+        return []byte{1}
+    }
+    return []byte{0}
+}
+
+func processMlDsaSigGen(vectors json.RawMessage, m Transactable) (interface{}, error) {
+	var groups []mlDsaSigGenTestGroup
+
+	if err := json.Unmarshal(vectors, &groups); err != nil {
+		return nil, err
+	}
+
+	var responses []mlDsaSigGenTestGroupResponse
+
+	for _, group := range groups {
+		if !strings.EqualFold(group.Type, "AFT") {
+			return nil, fmt.Errorf("unsupported sigGen test type: %v", group.Type)
+		}
+
+		response := mlDsaSigGenTestGroupResponse{
+			ID: group.ID,
+		}
+
+		for _, test := range group.Tests {
+			results, err := m.Transact("ML-DSA/"+group.ParameterSet+"/sigGen",
+			                            1, test.SK, test.Message, test.MU, test.RND, boolToBytes(group.ExternalMu))
+			if err != nil {
+				return nil, err
+			}
+
+			signature := results[0]
+
+			response.Tests = append(response.Tests, mlDsaSigGenTestCaseResponse{
+				ID:        test.ID,
+				Signature: signature,
+			})
+		}
+
+		responses = append(responses, response)
+	}
+	return responses, nil
+}
+
+type mlDsaSigVerTestGroup struct {
+	ID                 uint64 `json:"tgId"`
+	Type               string `json:"testType"`
+	ParameterSet       string `json:"parameterSet"`
+	Deterministic      bool   `json:"deterministic"`
+	SignatureInterface string `json:"signatureInterface"`
+	ExternalMu         bool   `json:"externalMu`
+	Tests        []struct {
+		ID        uint64               `json:"tcId"`
+		PK        hexEncodedByteString `json:"pk"`
+		Message   hexEncodedByteString `json:"message"`
+		MU        hexEncodedByteString `json:"mu"`
+		Signature hexEncodedByteString `json:"signature"`
+		Context   hexEncodedByteString `json:"context"`
+        HashAlg   string               `json:"hashAlg"`
```

Comment:
Be sure to run `go fmt` in the `acvptool` directory and it should take care of keeping the formatting consistent.

Reply:
Thanks, didn't know about that! done in [54906f6](https://github.com/aws/aws-lc/pull/2127/commits/54906f68f7cdbb6572e5a5da6afd5f6e1e36d1ab)